### PR TITLE
fix the bug of mkldnn zero_ when mkldnn tensor has a block format

### PR DESCRIPTION
**Pull request by XiaobingSuper**
MKLDNN has block format and the origin zero_ method can only set a part of memory value if a MKLDNN tensor has a block format(may be has [Padded format),](https://intel.github.io/mkl-dnn/understanding_memory_formats.html) so this PR solve this problem.

### CODE REVIEW

**Comment by XiaobingSuper** [comment]
@mingfeima, @Jianhui-Li, @jgong5, @gujinghui, @uyongw, @bddppq

File: test/test_mkldnn.py
Comment (bddppq) [inline test/test_mkldnn.py]: Could you add a comment here says mkldnn conv2d's weight is in block format?

File: test/test_mkldnn.py
Comment (bddppq) [inline test/test_mkldnn.py]: why does it need to go through save and load roundtrip?

Comment (XiaobingSuper) [reply test/test_mkldnn.py]: After using torch.jit.save, we can reorder mkldnn tensor from plain fromat to bolck format(call torch._C._nn.mkldnn_reorder_conv2d_weight function), please see https://github.com/pytorch/pytorch/blob/a6fb6e1fb39f1f4a509dec04e4719ff9e2d04c83/torch/utils/mkldnn.py#L60

File: test/test_mkldnn.py
Comment (XiaobingSuper) [reply test/test_mkldnn.py]: this step only get plain format(just call cpu_tensor.to_mkldnn()).

Comment (XiaobingSuper) [reply test/test_mkldnn.py]: just run this step: https://github.com/pytorch/pytorch/blob/a6fb6e1fb39f1f4a509dec04e4719ff9e2d04c83/torch/utils/mkldnn.py#L47

**Review by bddppq — APPROVED** [review]

**Comment by bddppq** [comment]
`fill_all_elems` is a new mkldnn api. When mkldnn was upgraded on github in https://github.com/pytorch/pytorch/pull/22910, the version in facebook internal codebase was not updated. We need to update it before being able to land this one. cc @VitalyFedyunin

**Comment by XiaobingSuper** [comment]
@VitalyFedyunin

**Comment by yf225** [comment]
Rebasing since this PR is more than 7 days old.

**Comment by vincentqb** [comment]
Rerunning internal tests as many are failing.

**Comment by XiaobingSuper** [comment]
@yf225 @bddppq

**Comment by XiaobingSuper** [comment]
@VitalyFedyunin

**Comment by XiaobingSuper** [comment]
@VitalyFedyunin, @bddppq

**Comment by XiaobingSuper** [comment]
@vincentqb

**Comment by vincentqb** [comment]
> `fill_all_elems` is a new mkldnn api. When mkldnn was upgraded on github in #22910, the version in facebook internal codebase was not updated. We need to update it before being able to land this one. cc @VitalyFedyunin

As mentioned above, I can't merge the diff since there are many internal errors.

**Pull request closed without merging**
